devito/nodes.py

```diff
@@ -80,20 +86,50 @@ def __init__(self, header=None, body=None, footer=None):
 
     def __repr__(self):
         header = "".join([str(s) for s in self.header])
-        body = "Block::\n\t%s" % "\n\t".join([str(s) for s in self.body])
+        body = "\n\t".join([str(s) for s in self.body])
         footer = "".join([str(s) for s in self.footer])
-        return header + body + footer
+        return "%s::\n\t%s" % (self.__class__.__name__, header + body + footer)
 
     @property
     def ccode(self):
         body = tuple(s.ccode for s in self.body)
-        return c.Block(self.header + body + self.footer)
+        return self._wrapper(self.header + body + self.footer)
 
     @property
     def children(self):
         return (self.body,)
 
 
+class List(Block):
+
+    """Class representing a sequence of one or more statements."""
+
+    is_List = True
+
+    _wrapper = c.Collection
+
+
+class Element(Node):
+
+    """A generic node that is worth identifying in an Iteration/Expression tree.
+
+    It corresponds to a single :class:`cgen.Statement`.
```

Comment:
In that case why not mirror the naming and call it `Statement`? "Element" is a fairly overloaded term, but I'm really just being pedantic here.